src/conversion/model-to-view-converters.js

```diff
@@ -288,6 +289,157 @@ export function unwrap( elementCreator ) {
+ *
+ *		modelDispatcher.on( 'removeMarker:searchResult', unwrapMarker( new ViewAttributeElement( 'span', { class: 'searchResult' } ) ) );
+ *
+ * @see engine.conversion.modelToView.wrapMarker
+ * @external engine.conversion.modelToView
+ * @function engine.conversion.modelToView.unwrapMarker
+ * @param {engine.view.AttributeElement|Function} elementCreator View attribute element, or function returning
+ * a view attribute element, which will be used for unwrapping.
+ * @returns {Function} Remove attribute converter.
+ */
+export function unwrapMarker( elementCreator ) {
+	return ( evt, data, consumable, conversionApi ) => {
+		if ( !consumable.consume( data.range, 'range' ) ) {
+			return;
+		}
+
+		const viewElement = ( elementCreator instanceof ViewElement ) ?
+			elementCreator.clone( true ) :
+			elementCreator( data, consumable, conversionApi );
+
+		if ( data.range.isCollapsed ) {
+			const modelStart = data.range.start;
+			let viewStart = conversionApi.mapper.toViewPosition( modelStart );
+
+			if ( viewStart.parent instanceof ViewText ) {
+				viewStart = ViewPosition.createAfter( viewStart.parent );
+			}
+
+			const viewEnd = findFurthestZeroLengthPoint( viewStart, conversionApi.mapper );
+
+			viewWriter.unwrap( new ViewRange( viewStart, viewEnd ), viewElement );
+		} else {
+			const viewRange = conversionApi.mapper.toViewRange( data.range );
+			const flatViewRanges = breakViewRangeOnFlat( viewRange );
+
+			for ( let range of flatViewRanges ) {
+				viewWriter.unwrap( range, viewElement );
+			}
+		}
+	};
+}
+
+// Starting from given `viewPosition`, moving forward, finds the furthest position in a view, which is
+// separated only by elements with zero model length (which in most cases are empty `ViewAttributeElement`s).
+// Using `viewPosition` and returned position, one can build a view range, which start and position are mapped to
+// the same position in model.
+// @param {engine.view.Position} viewPosition Position to start from when looking for furthest zero length position.
+// @param {engine.conversion.Mapper} mapper Mapper to use when looking for furthest zero length position.
+// @returns {engine.view.Position} Furthest zero length position.
+function findFurthestZeroLengthPoint( viewPosition, mapper ) {
+	viewPosition = ViewPosition.createFromPosition( viewPosition );
+
+	while ( viewPosition.nodeAfter && mapper.getModelLength( viewPosition.nodeAfter ) === 0 ) {
+		viewPosition.offset++;
+	}
+
+	return viewPosition;
+}
+
+// Breaks given `viewRange` on "flat" view ranges, where "flat" means that the range is contained within
+// one container element. After `viewRange` is broken, it's "pieces" can be used in ViewWriter (which expects
+// that passed ranges are contained within one container element).
+// @param {engine.view.Range} viewRange Range to break.
+// @returns {Array.<engine.view.Range>} Flat ranges that combines into passed `viewRange`.
+function breakViewRangeOnFlat( viewRange ) {
```

Comment:
For now, I will just move it. I thought about making this a default behavior of methods like `viewWriter.remove` or `viewWriter.wrap` but those methods return values and I think it would be too many changes in this PR.